### DISCUSSION TITLE
CPU Tuning for a Zenbook 14 OLED (Ryzen 5900HX)

### DISCUSSION THREAD

**Discussion by JoseHidalgo** [seerge/g-helper#1678] [Q&A] [answered]
Hi everybody. I have a brand new Zenbook 14 Flip OLED (5900HX), and I have just installed G-Helper.

My main goal is to lower power consumption and increase battery runtime:

- The OLED screen already draws some power, but lowering brightness keeps it under control (plus it's good for the screen). G-Helper says it's set to "90Hz + OD". I guess I'd like to keep it that way for visual reasons (fluidity, etc).
- Most importantly, the 5900HX CPU is too powerful for my taste. It supposedly has a TDP of 45+ W, which is too much for my basic use: a lot of Internet, MS Office, Spotify, some 1080p movies with MPV... No gaming and nothing CPU-intensive that I can think of.

According to reviews [such as this one](https://www.notebookcheck.net/Asus-Zenbook-Flip-14-OLED-review-An-absolute-sensation-thanks-to-AMD-and-OLED.620511.0.html) (see the Power Consumption section), the 5900HX tends to run too hot, which makes the laptop hot and makes the fans too loud. My goal is to keep everything cool and discrete.

G-Helper offers some CPU settings I'm not used to, so I could use a hand to find those that have the best impact on my 5900HX:

- CPU Boost: default is "Aggressive". There are 7 settings.
- Power Limits: the slider is set by default to 125W (CPU Slow (SPL + sPPT). The "apply limits" checkbox is not checked.

I have no idea of what all that means. I'd just like to tame my 5900HX within reasonable limits and increase battery runtime, which would allow me to limit battery charge to 60% and also increase battery life. What would you do, please?

Thanks in advance for your help!

**Comment by seerge** [comment]
@JoseHidalgo hello,

1. CPU boost setting is just a shortcut to corresponding setting in windows power plan in control panel https://github.com/seerge/g-helper/discussions/492
2. When power limits ``Apply`` is not checked, your system will use default ones for each mode (hardcoded in BIOS). Unless you have issues I wouldn't restrict them anyhow, as your system won't drain more power than needed :)

**Comment by JoseHidalgo** [comment]
Thank you @seerge , much appreciated :)

With that said, I'm still unsure of what I should do:

- I've read that limiting CPU boost could lower power consumption significantly on some CPUs. I don't know about the 5900HX though.
- Other people have suggested that the power limit of the 5900HX could be lowered from 45W to 15W (but didn't say how).

You say the system won't drain more power than needed, but this particular convertible seems to be known for running quite hot and not being particularly battery-efficient (see the review that I linked).

My point is: I bought this laptop because it was a good Black Friday deal, but its 45W CPU is overkill for me. I was aiming to buy an ultra-low consumption convertible, with some "U" 15W CPU and a huge battery runtime (some of those convertibles get about 20h on battery). Instead I got a convertible with about 8 hours of battery runtime. But if I want to keep my battery between, say, 20 and 60%, then my runtime becomes insufficient (less than 4 hours). That's what I'd like to increase.

Of course a good part of that is **idling**. The OLED draws power, but like I said reducing brightness helps. Now I only need to take care of the 5900HX CPU and make sure that it draws as little power as possible. Unless some day I do something exceptionally power-consuming, but that day I can always use G-Helper to remove limits and get back to normal.

I hope you see my point :)

**Reply by Kikimahe** [reply, marked as answer]
Hi,

Maybe I'm going to say something stupid, but why not just modify the balanced Windows power plan ?

In processor power management you can define the performance of your processor on battery and plugged in
By reducing the performance percentage you reduce consumption.
![zFcgE](https://github.com/seerge/g-helper/assets/142881210/f386fc11-a541-4252-9157-4866916f1b86)



Otherwise use Ghelper to limit CPU Power :
1) Disable CPU Boost
2) Limit your processor to 15W if you wish and check “apply limits”

With GHelper you can even have different power limits in each CPU Mode, silent, balanced or Turbo.

![Screenshot 2023-12-01 151754](https://github.com/seerge/g-helper/assets/142881210/7c2e131c-2c0f-47c4-b9a6-5ea737258a0f)

You can also monitor the power consumption of your CPU with Hwinfo64 for example. Before and after the changes.
![01](https://github.com/seerge/g-helper/assets/142881210/53cd66c9-d879-43a6-a65c-99d052139ade)


 :-)

**Comment by JoseHidalgo** [comment]
Oh wow, I hadn't even thought of that! Thank you @Kikimahe! (BTW I'm french too, lol)
BTW I guess you have a similar concern with your own laptop: the 12700H is a beast but it can draw a lot of power!

I would like to do a mix of your two suggestions:

1. Modify the power plan, to reduce CPU performance only on battery mode
2. I'd also like to disable CPU Boost... but only on battery mode. Can G-Helper do that? From what I see, if I disable CPU Boost, it will be disabled permanently, not only on battery.

**EDIT - I've found a way to disable CPU Boost right from Windows Power Options.**
It's a Windows feature that's hidden by default. We just have to make it appear magically.
The Registry hack is here: https://www.reddit.com/r/ZephyrusG14/comments/gho535/important_update_to_properly_disable_boosting/
Works perfectly. So my new settings are:
- On battery : CPUmin 5% / CPUmax 50% / CPUBoost OFF
- On power: CPUmin 80% (default) / CPUmax 100% (default) / CPUBoost Aggressive-Efficient (from what I've read in the reddit topic, it's about the same as Aggressive but like its name suggests, more efficient)

The good news is that I hope these settings will help me to maximize my battery runtime.
The bad news is that I didn't need to use G-Helper for that! ;p 
But if these settings are not enough, I can always come back to G-Helper and cap the CPU to 15W max.
We'll see how it goes.

**Reply by Kikimahe** [reply]
Yes I'm French and I racked my brains to find screenshots in English... :D

To your point 2. Ghelper can do it !
You should know that Ghelper remembers the CPU mode (silent, balanced, Turbo) that you choose when you are on battery.
And in each of the 3 modes, you can define the CPU Boost parameter independently of each other.

So On battery, you can select silent mode with CPU Boost disabled:
- When you are on battery, Ghelper will automatically switch to silent mode with CPU Boost disabled.
- When you plug in, Ghelper will automatically switch to Balanced or Turbo mode with CPU Boost enable.

Ghelper is a great tool and Seerge is an artist. :-)

**Comment by JoseHidalgo** [comment]
Nice! Will try that! :D

Merci beaucoup @Kikimahe, and of course thanks to Seerge for designing such a great piece of software!
